plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/ExamplePluginConfiguration.java

```diff
@@ -19,51 +19,39 @@
 
 package org.elasticsearch.plugin.example;
 
-import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.yaml.YamlXContent;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 
 import java.io.IOException;
 import java.nio.file.Path;
-
-import static java.nio.charset.StandardCharsets.UTF_8;
-import static java.nio.file.Files.newBufferedReader;
-import static org.elasticsearch.common.io.Streams.copyToString;
+import java.util.Locale;
 
 /**
  * Example configuration.
  */
 public class ExamplePluginConfiguration {
-    private String test = "not set in config";
-
+    
+    private final Settings settings;
+    
+    private final Setting<String> TEST_SETTING;
+    
     @Inject
     public ExamplePluginConfiguration(Environment env) throws IOException {
         // The directory part of the location matches the artifactId of this plugin
-        Path configFile = env.configFile().resolve("jvm-example/example.yaml");
-        String contents = copyToString(newBufferedReader(configFile, UTF_8));
-        XContentParser parser = YamlXContent.yamlXContent.createParser(contents);
-
-        String currentFieldName = null;
-        XContentParser.Token token = parser.nextToken();
-        assert token == XContentParser.Token.START_OBJECT;
-        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            if (token == XContentParser.Token.FIELD_NAME) {
-                currentFieldName = parser.currentName();
-            } else if (token.isValue()) {
-                if ("test".equals(currentFieldName)) {
-                    test = parser.text();
-                } else {
-                    throw new ElasticsearchParseException("Unrecognized config key: {}", currentFieldName);
-                }
-            } else {
-                throw new ElasticsearchParseException("Unrecognized config key: {}", currentFieldName);
-            }
-        }
+        Path path = env.configFile().resolve("jvm-example/example.yaml");
+        settings = Settings.builder().loadFromPath(path).build();
+       
+        // asserts for tests
+        assert settings != null;
+        assert settings.get("test") != null;
+
+        String testValue = settings.get("test", "default_value");
+        TEST_SETTING = new Setting<String>("test", testValue, (value) -> value.toLowerCase(Locale.ROOT), Setting.Property.Dynamic);   
```

Comment:
Instead of creating the setting here, it should be a static setting defined in the class, similar to the way settings are defined in https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java#L37